test/serialize.test.ts

```diff
@@ -0,0 +1,71 @@
+import { SRPParameters } from "../src/parameters";
+import { SRPRoutines } from "../src/routines";
+import {
+  SRPClientSession,
+  SRPClientSessionStep1,
+  SRPClientSessionStep2,
+} from "../src/session-client";
+import { SRPServerSession, SRPServerSessionStep1 } from "../src/session-server";
+import { createVerifierAndSalt, generateRandomString } from "../src/utils";
+import { test } from "./tests";
+
+const TEST_ROUTINES = new SRPRoutines(new SRPParameters());
+
+test("#SRP serialization", async (t) => {
+  const testUsername = generateRandomString(10);
+  const testPassword = generateRandomString(15);
+
+  const { s: salt, v: verifier } = await createVerifierAndSalt(
+    TEST_ROUTINES,
+    testUsername,
+    testPassword,
+  );
+
+  const serverStep1 = await new SRPServerSession(TEST_ROUTINES).step1(
+    testUsername,
+    salt,
+    verifier,
+  );
+
+  const serializedServerStep1 = JSON.stringify(serverStep1); // internally calls .toJSON
+
+  // call fromState to deserialize and resume state
+  const deserializedServerStep1 = SRPServerSessionStep1.fromState(
+    TEST_ROUTINES,
+    JSON.parse(serializedServerStep1),
+  );
+
```

Comment:
Should we add some test to make sure serverStep1 is the same as deserializedServerStep1?

Reply:
you can check with e.g. `(step1 as any).privateValue`, as `private` is only at TS level, you can access it at runtime.

Reply:
re: necessity, I am not sure. the test covers that you can serde between steps, and there is some random generated key, that should be fine. Inspecting private/inner state is better avoided.